spark/src/main/java/org/opensearch/sql/spark/client/EmrClientImpl.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.client;
+
+import static org.opensearch.sql.spark.data.constants.SparkConstants.FLINT_INTEGRATION_JAR;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.SPARK_APPLICATION_JAR;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.SPARK_INDEX_NAME;
+import static org.opensearch.sql.spark.data.constants.SparkConstants.STEP_ID_FIELD;
+
+import com.amazonaws.services.elasticmapreduce.model.ActionOnFailure;
+import com.amazonaws.services.elasticmapreduce.model.AddJobFlowStepsRequest;
+import com.amazonaws.services.elasticmapreduce.model.AddJobFlowStepsResult;
+import com.amazonaws.services.elasticmapreduce.model.DescribeStepRequest;
+import com.amazonaws.services.elasticmapreduce.model.HadoopJarStepConfig;
+import com.amazonaws.services.elasticmapreduce.model.StepConfig;
+import com.amazonaws.services.elasticmapreduce.model.StepStatus;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import lombok.SneakyThrows;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONObject;
+import org.opensearch.client.Client;
+import org.opensearch.sql.spark.helper.EMRHelper;
+import org.opensearch.sql.spark.helper.FlintHelper;
+import org.opensearch.sql.spark.response.SparkResponse;
+
+public class EmrClientImpl implements SparkClient {
+  private final Client client;
+  private final EMRHelper emr;
+  private final FlintHelper flint;
+  private final String field = STEP_ID_FIELD;
+  private static final Logger logger = LogManager.getLogger(EmrClientImpl.class);
+
+  /**
+   * Constructor for EMR Client Implementation.
+   *
+   * @param client  Opensearch client
+   * @param emr     EMR helper
+   * @param flint   Opensearch args for flint integration jar
+   */
+  public EmrClientImpl(
+      Client client, EMRHelper emr, FlintHelper flint) {
+    this.client = client;
+    this.emr = emr;
+    this.flint = flint;
+  }
+
+  @Override
+  public JSONObject sql(String query) throws IOException {
+    return new SparkResponse(
```

Comment:
Is SparkResponse helper class?
If yes, make it static.

Reply:
setting stepId afterwards so can't make it static.